src/CoreWCF.Primitives/src/CoreWCF/Channels/TransportSecurityBindingElement.cs

```diff
@@ -0,0 +1,216 @@
+                    securityServiceDispatcher.SessionServerSettings.TolerateTransportFailures = this.LocalServiceSettings.ReconnectTransportOnFailure;
+                    securityServiceDispatcher.SessionServerSettings.CanRenewSession = scParameters.CanRenewSession;
+                    securityServiceDispatcher.SessionServerSettings.IssuedSecurityTokenParameters = scParameters.Clone();
+                    ((SecureConversationSecurityTokenParameters)securityServiceDispatcher.SessionServerSettings.IssuedSecurityTokenParameters).IssuerBindingContext = issuerBindingContext;
+                    securityServiceDispatcher.SessionServerSettings.SecurityStandardsManager = sessionFactory.StandardsManager;
+                    securityServiceDispatcher.SessionServerSettings.SessionProtocolFactory = sessionFactory;
+                    securityServiceDispatcher.SecurityProtocolFactory = (SecurityProtocolFactory)sessionFactory;
+
+                    // pass in the error handler for handling unknown security sessions - dont do this if the underlying channel is duplex since sending 
+                    // back faults in response to badly secured requests over duplex can result in DoS.
+                    //if (context.BindingParameters != null && context.BindingParameters.Find<IChannelDemuxFailureHandler>() == null
+                    //    && !IsUnderlyingListenerDuplex<TChannel>(context))
+                    //{
+                    //    context.BindingParameters.Add(new SecuritySessionServerSettings.SecuritySessionDemuxFailureHandler(sessionFactory.StandardsManager));
+                    //}
+                }
+                else
+                {
+                    //TODO later 
+                    //TransportSecurityProtocolFactory protocolFactory = new TransportSecurityProtocolFactory();
+                    //base.ApplyAuditBehaviorSettings(context, protocolFactory);
+                    //this.EndpointSupportingTokenParameters.Endorsing.RemoveAt(0);
+                    //try
+                    //{
+                    //    base.ConfigureProtocolFactory(protocolFactory, credentialsManager, true, issuerBindingContext, context.Binding);
+                    //    SecureConversationSecurityTokenParameters acceleratedTokenParameters = (SecureConversationSecurityTokenParameters)scParameters.Clone();
+                    //    acceleratedTokenParameters.IssuerBindingContext = issuerBindingContext;
+                    //    protocolFactory.SecurityBindingElement.EndpointSupportingTokenParameters.Endorsing.Insert(0, acceleratedTokenParameters);
+                    //}
+                    //finally
+                    //{
+                    //    this.EndpointSupportingTokenParameters.Endorsing.Insert(0, scParameters);
+                    //}
+
+                    //channelListener.SecurityProtocolFactory = protocolFactory;
+                }
+
+            }
+            //else
+            //{
```

Comment:
This is also an important scenario. This is when the client isn't using secure conversation and the credential security token is included in every request.

Reply:
incorporated...let's add test too..